recipes/asyncqt/meta.yaml

```diff
@@ -0,0 +1,40 @@
+{% set name = "asyncqt" %}
+{% set version = "0.7.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8b1507c968c85cf0b7eee5d2a887162d38af15fb8c5b1ec25beed6025d7383ac
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
+    - setuptools
+  run:
+    - python >=3.5
+
+test:
+  requires:
+    - pyside2
```

Comment:
Bummer, both `PyQt5` and `PySide2` fail on import:

```
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

@marcelotrevisani any hints on how to proceed?

Reply:
uunn.... I know how one way to fix it. However, this recipe will not be able to be ``noarch``. 
You can add the libraries for linux related to the ``libgl``.
Can you test it in another way?

Reply:
One thing which I was wondering is, does this recipe rely on ``qt``? I mean, is ``qt`` a dependency for this library?

Reply:
> Can you test it in another way?

I can't think of how to test it another way without importing it...

> I was wondering is, does this recipe rely on `qt`?

Not explicitly, it will pick up and work with `PyQt5` or `PySide`, whatever is installed. See: https://github.com/gmarull/asyncqt/blob/fa3f466e7060928fef64c600c8ccebe60045bad2/asyncqt/__init__.py#L34-L51

> You can add the libraries related to `libgl` for linux platform.

We can make it non-arch then, if it means we can at least test it. Which platform do we need to install in Linux, Windows and OS-X to get GL support?

Reply:
I read the documentation and it seems that, ``pyside``/``pyqt`` are ``run requirements``.

I think you will have to add ``xorg-libx11 # [linux]`` for linux platform

Reply:
For windows and osx that should not be a problem, as you can see the package was successfully created for those

Reply:
> I read the documentation and it seems that, `pyside`/`pyqt` are `run requirements`.

Yes, to use the package you need either PySide2 or PyQt5 installed, that is correct. But we should *not* require both of them to be installed when `asyncqt` is installed, otherwise users will install both PyQt5 and PySide2 at the same time which is definitely not what one wants. 

> I think you will have to add `xorg-libx11 # [linux]` for linux platform

Thanks!

Reply:
Hmmm still the same error on Linux:

```
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

Any other ideas?

Reply:
Btw, just remembered that conda supports [constrained dependencies](https://github.com/conda/conda/issues/1696#issuecomment-300309736) since 4.4, although that has not been documented yet (https://github.com/conda/conda-docs/issues/580):


```yaml
  run:
    - python >=3.5
  run_constrained:
    - pyside2
    - pyqt 5
```